app/controllers/managers_controller.rb

```diff
@@ -109,12 +107,33 @@ def destroy
     render :json => @manager.to_json
   end
 
+  def add_city
+    @manager = Manager.find(params[:id])
+    CommunityLeadership.where(manager_id: @manager.id, city_id: params[:city_id]).first || CommunityLeadership.create(manager_id: @manager.id, city_id: params[:city_id])
```

Comment:
`CommunityLeadership.create` might fail. Consider handling this case.

Reply:
done